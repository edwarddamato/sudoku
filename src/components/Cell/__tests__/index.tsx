jest.mock('../../../store', () => ({
  Store: {
    GetGridValues: () => ({
      values: [
        [0, 0, 0, 0, 0, 0, 0, 0, 0],
        [0, 1, 0, 0, 0, 0, 0, 0, 0],
        [0, 0, 2, 0, 0, 0, 0, 0, 0],
        [0, 0, 0, 3, 0, 0, 0, 0, 0],
        [0, 0, 0, 0, 4, 0, 0, 0, 0],
        [0, 0, 0, 0, 0, 5, 0, 0, 0],
        [0, 0, 0, 0, 0, 0, 6, 0, 0],
        [0, 0, 0, 0, 0, 0, 0, 7, 0],
        [0, 0, 0, 0, 0, 0, 0, 0, 8]
      ]
    }),
    Subscribe: () => ({})
  }
}));

import * as React from 'react';
import { Cell } from '../index';
import { create } from 'react-test-renderer';

it('renders correctly', () => {
  const cell = create(<Cell blockIndex={5} index={5} onChange={() => ({})} />).toJSON();
  expect(cell).toMatchSnapshot();
});
