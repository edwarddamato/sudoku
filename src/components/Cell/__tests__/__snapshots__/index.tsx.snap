// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="component_game-cell"
>
  <input
    className="game-cell_input"
    maxLength={1}
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    placeholder="0"
    type="text"
    value="5"
  />
</div>
`;
